docs/extras/use_cases/web_scraping/web_scraping_with_openai_functions.ipynb

```diff
@@ -0,0 +1,389 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "6605e7f7",
+   "metadata": {},
+   "source": [
+    "# Web scraping using OpenAI Functions Extraction chain\n",
+    "\n",
+    "Web scraping is challenging for many reasons; one of them is the changing nature of modern websites' layouts and content, which requires modifying scraping scripts to accommodate the changes.\n",
+    "\n",
+    "Using OpenAI Function in the extraction chain, you can avoid having to change your code constantly when websites change. Plus, this will save you time setting up when scraping a website for the first time by creating simple schemas.\n",
+    "\n",
+    "In this notebook, we scrape The Wall Street Journal. You'll need to have an OpenAI API key for this example."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "05087ad3",
+   "metadata": {},
+   "source": [
+    "## Install dependencies\n",
+    "\n",
+    "We need OpenAI and LangChain for the extraction part, and Playwright and Beautiful Soup for the scraping part."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "00c96747",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m23.0.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m23.2.1\u001b[0m\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpython3.11 -m pip install --upgrade pip\u001b[0m\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "pip install -q openai langchain playwright beautifulsoup4"
```

Comment:
Also need to add:

```
! playwright install
```

to download the necessary browser binaries (Chromium, Firefox, WebKit).